api/tokens.go

```diff
@@ -148,3 +151,72 @@ func (api *API) RefreshHandler(ctx context.Context, w http.ResponseWriter, req *
 
 	return models.NewSuccessResponse(jsonResponse, http.StatusCreated, headers), nil
 }
+
+//SignOutAllUsersHandler bulk refresh token invalidation for panic sign out handling
+func (api *API) SignOutAllUsersHandler(ctx context.Context, w http.ResponseWriter, req *http.Request) (*models.SuccessResponse, *models.ErrorResponse) {
+	usersList := models.UsersList{}
+	listUserInput := usersList.BuildListUserRequest("status=\"Enabled\"", "username", int64(0), &api.UserPoolId)
+	listUserResp, err := api.CognitoClient.ListUsers(listUserInput)
+	if err != nil {
+		return nil, models.NewErrorResponse(http.StatusInternalServerError, nil, err)
+	}
+	usersList.MapCognitoUsers(listUserResp)
+	globalSignOut := &models.GlobalSignOut{
+		ResultsChannel: make(chan string, len(usersList.Users)),
+		BackoffSchedule: []time.Duration{
+			1 * time.Second,
+			3 * time.Second,
+			10 * time.Second,
+		},
+		RetryAllowed: true,
+	}
+	// run api.SignOutUsersWorker concurrently
+	go api.SignOutUsersWorker(ctx, globalSignOut, &usersList.Users)
```

Comment:
Firing off a goroutine without knowing how or when it will end seems like potentially undefined behaviour to me.. was there a reason this was done concurrently?

Reply:
The decision was taken to run this go routine in non-blocking asynchronous mode - and let it handle the calls to to the api without the caller having to handle any returns/errors.